components/ModSubmissionsEditFormFacility.vue

```diff
@@ -341,7 +346,121 @@ function submitForm(e: Event) {
+
+    if (!updateSubmissionId) {
+        console.error('Facility ID is required for updating the facility')
+        return
+    }
+
+    const facilityInputVariables = {
+        updateSubmissionId,
+        input: {
+            isUnderReview: true,
+            facility: {
+                nameEn: nameEn.value || '',
+                nameJa: nameJp.value || '',
+                contact: {
+                    googleMapsUrl: googlemapsURL.value || '',
+                    email: email.value || '',
+                    phone: phone.value || '',
+                    website: website.value || '',
+                    address: {
+                        postalCode: postalCode.value || '',
+                        prefectureEn: prefectureEn.value || '',
+                        cityEn: cityEn.value || '',
+                        addressLine1En: addressLine1En.value || '',
+                        addressLine2En: addressLine2En.value || '',
+                        prefectureJa: prefectureJp.value || '',
+                        cityJa: cityJp.value || '',
+                        addressLine1Ja: addressLine1Jp.value || '',
+                        addressLine2Ja: addressLine2Jp.value || ''
+                    }
+                },
+                healthcareProfessionalIds: [],
+                mapLatitude: parseFloat(mapLatitude.value) || 0,
+                mapLongitude: parseFloat(mapLongitude.value) || 0
+            }
+        }
+    }
+
+    const maxAttempts = 3
+    let mutationAttempts = 0
+    let mutationSuccess = false
+
+    function delay(milliseconds: number) {
+        return new Promise(resolve => setTimeout(resolve, milliseconds))
+    }
+
+    while (mutationAttempts < maxAttempts && !mutationSuccess) {
+        try {
+            moderationSubmissionStore.handleUpdateIsSendingUpdatedForm(true)
+            await gqlClient.request(updateFacilitySubmissionGqlQuery, facilityInputVariables)
+            moderationSubmissionStore.handleUpdateIsSendingUpdatedForm(false)
+            moderationSubmissionStore.handleDidMutationFail(false)
+            mutationSuccess = true
+            if (moderationSubmissionStore.updateSubmissionFromTopNav) {
+                moderationSubmissionStore.handleUpdateSubmissionFromTopNav(false)
+                router.push('/moderation')
+            }
+        } catch (error) {
+            console.error(`Attempt ${mutationAttempts + 1} failed:`, error)
+            if (mutationAttempts < maxAttempts) {
+                moderationSubmissionStore.handleDidMutationFail(true)
+                await delay(60000) // 60 seconds delay
+                mutationAttempts++
+            }
+        }
+        if (mutationAttempts === maxAttempts) {
+            moderationSubmissionStore.handleUpdateIsSendingUpdatedForm(false)
+            moderationSubmissionStore.handleUpdateSubmissionFromTopNav(false)
+            moderationSubmissionStore.handleDidMutationFail(false)
+        }
+    }
 }
+
```

Comment:
this sets up a synthetic event to mock the submit button event when the state changes on click from the Top Nav to save and exit